### DISCUSSION TITLE
Remove obsolete code in modern scenarios?

### DISCUSSION THREAD

**Discussion by LevT2** [Microsoft/WSLab#530] [Ideas]
Why don't you fork MSLab code (and appreciate) removing old workarounds and build checks?

As a whole, it is literally incomprehensible for newcomers now. I doubt any user would invest in running complex older scenarios in hurry, while it is challenging to grasp the current state of the codebase and accustom its usage.

Meanwhile, I'm impressed by the authors' effort and overall code quality and best practiced followed (unfortunately, except for the being late with modularization or similar). I am still really trying to comprehend the codebase to be able to contribute something, instead of copy-pasting code chunks into my own scripts. So, I would appreciate if you froze the current build logic and fork a new debt-free development.

**Comment by machv** [comment]
It's a good point that for new users this might look unstructured, but I still find even older scenarios useful enough to keep them in repository just for the reference. 

One of the options I was considering was to split repository into two separate repositories: one only with MSLab scripts, and another one with scenarios only (in addition to https://www.geos.to/azslabs), but in the end, it think this split would just complicate things.

What might be useful is to generate a list with all the scenarios with Last updated date, which would give an indication how up to date the content is.

**Comment by LevT2** [comment]
@machv 
  Thanks for your reply. I totally agree, all the scenarios retain their value, at least as an excellent scripting and programmability reference. 

I proposed forking the infrastructure: it's critical to my comprehension not to mess with old OS and build checks and obsolete workarounds in Deploy.ps1 and CreateParentDisk.ps1, and not to have the latest doubled (as 2_ prepended temporary script and as the downloadable "prerequisite"). Seemingly, I also found chunks of duplicate functionality between Convert-WindowsImage.ps1 helper functions (they are being dotsourced as well) and your code - but I still can't convince myself in it, as I just don't grasp the whole.

**Reply by jaromirk** [reply]
If you will be exploring some older scenario, let me know. Most are still relevant. But we can discuss it here.
